### ISSUE TITLE
[Feature Request]: CLS: add global constants/functions to completion suggestions

### ISSUE THREAD

**Issue by ninivert** [chapel-lang/chapel#24743]
### Summary of Feature

When starting from an empty file, `chpl-language-server` should show the globally available symbols when prompted for suggestions, for instance:

- `numLocales`, `Locales`, `here`, ...
- `writeln`, `write`, `warning`, ...
- probably others that I'm not aware of

In the VSCode extension, some of these are made available through snippets: https://github.com/chapel-lang/chapel-vscode/blob/main/snippets/chapel.json

### Code Sample

```chapel
writeln(numLocales, " should equal ", Locales.size);
write("hello world", '--- \n');
```

Writing the above code would be much faster with this feature.

Currently `Locales.`\<next LSP suggestion\> yields nothing from CLS, where I would expect methods to be available on the array of locales.

**Comment by ninivert** [comment]
Hi !

Indeed, this can be seen as two incremental parts. Just having the global symbols would be very useful to beginners in the language (such as me :smiley:).

I initially thought CLS was able to generate suggestions based on type information, for instance with the following program,

``` chapel
const D = {1..10};
var A: [D] real;
A.<next LSP suggestion>
```

CLS could propose `rank`, `strides`, `size`, etc. (from https://chapel-lang.org/docs/language/spec/arrays.html#predefined-routines-on-arrays), based on the fact it's an array (of reals). Unless I've messed up in setting up CLS, it seems this isn't yet available (so as you say, this would be the scope of another feature request)

**Comment by DanilaFe** [comment]
Oops, I didn't mean to auto-close this. @ninivert -- we just recently merged a PR that addresses the 'global symbols' part of your issue (`Locales`, `numLocales`, etc.). We haven't yet gotten support for `Locale.size` etc, since that's significantly more complicated, as we have discussed.

Since this original issue is largely about global symbols, I would be inclined to close it, and open another for methods / fields of types (i.e., `Locales.size`, and really `anything.anythingelse`). Does that sound reasonable to you?
